rcl_action/src/rcl_action/action_server.c

```diff
@@ -319,6 +319,9 @@ rcl_action_server_get_default_options(void)
   RCL_CHECK_ARGUMENT_FOR_NULL(response, RCL_RET_INVALID_ARGUMENT); \
   rcl_ret_t ret = rcl_send_response( \
     &action_server->impl->Type ## _service, response_header, response); \
+  if (RCL_RET_TIMEOUT == ret) { \
+    return RCL_RET_TIMEOUT;  /* error already set */ \
```

Comment:
adds error return code RCL_RET_TIMEOUT from the service call, and client libraries can handle this timeout case.